### PR TITLE
Agent subscription

### PR DESCRIPTION
**Pull request by suyangyu**
added parameter enroluser to ferequest and updated rosm service to call enrolment only if this param is set to true

### CODE REVIEW

File: app/auth/Authenticated.scala [hunk 1]
```diff
@@ -35,7 +37,16 @@ trait Authenticated {
...
+  def authenticated(f: => Future[Result])(implicit hc: HeaderCarrier): Future[Result] =
+    auth.getCurrentAuthority().flatMap { authority =>
+      Logger.debug(s"Got authority = $authority")
+      mapToAuthResult(authority) match {
+        case NotLoggedIn => Unauthorized
+        case _ => f
+      }
+    }
+
+  def authenticatedCustom(f: => AuthenticationResult => Future[Result])(implicit hc: HeaderCarrier): Future[Result] = {
```
Comment (MJCallahanPage) [inline app/auth/Authenticated.scala:49]: What does authenticaatedCustom mean? Is there a better name we can find for this?

**Review by MJCallahanPage — CHANGES_REQUESTED** [review]
One comment re. naming.

**Comment by MJCallahanPage** [comment]
@suyangyu - can you resolve the expected conflicts please. Thanks

File: app/auth/Authenticated.scala (hunk 1 above)
Comment (suyangyu) [reply app/auth/Authenticated.scala:49]: not sure what to call this one, I'm open to suggestions...

Basically in this function you need to pass in a function which you must specify how to deal with the authentication results. Whereas the other function it has default behaviours.

The custom function is solely used for the throttling at the moment so I could call it something along the lines of throttleAuthentication?

Comment (MJCallahanPage) [reply app/auth/Authenticated.scala:49]: Hmmmm, I see. It's a tricky one then.

Leave it as it is then for  now

**Review by MJCallahanPage — APPROVED** [review]

**Pull request merged**